src/librustc_resolve/diagnostics.rs

```diff
@@ -354,6 +354,17 @@ impl<'a> Resolver<'a> {
                     span, "defaulted type parameters cannot be forward declared".to_string());
                 err
             }
+            ResolutionError::SelfInTyParamDefault => {
+                let mut err = struct_span_err!(
+                    self.session,
+                    span,
+                    E0735,
+                    "type parameters with a default cannot use `Self`"
+                );
+                err.span_label(
+                    span, "defaulted type parameters cannot use `Self`".to_string());
```

Comment:
```suggestion
                    span, "`Self` in type parameter default".to_string());
```
I think this is clearer.